scala/private/phases/phase_coverage.bzl

```diff
@@ -0,0 +1,78 @@
+#
+# PHASE: coverage
+#
+# DOCUMENT THIS
+#
+
+load(
+    "@io_bazel_rules_scala//scala/private:coverage_replacements_provider.bzl",
+    _coverage_replacements_provider = "coverage_replacements_provider",
+)
+
+def phase_coverage_library(ctx, p):
+    args = struct(
+        srcjars = p.collect_srcjars,
+    )
+    return _phase_coverage_default(ctx, p, args)
+
+def phase_coverage_common(ctx, p):
+    return _phase_coverage_default(ctx, p)
+
+def _phase_coverage_default(ctx, p, _args = struct()):
+    return _phase_coverage(
+        ctx,
+        p,
+        _args.srcjars if hasattr(_args, "srcjars") else depset(),
+    )
+
+def _phase_coverage(ctx, p, srcjars):
+    if len(ctx.files.srcs) + len(srcjars.to_list()) == 0 or \
+       not ctx.configuration.coverage_enabled or \
+       not hasattr(ctx.attr, "_code_coverage_instrumentation_worker"):
```

Comment:
can/should we drop this condition about the attribute? I think it made sense when it was coupled to the mandatory compile phase.
If a rule doesn't have this attribute it shouldn't use this phase. 
Moreover we want this rule to fail and make a conscious decision, right?

Reply:
This condition seems redundant since it's one of `common_attrs` so every rules except `scala_library_for_plugin_bootstrapping` should have it.

How about removing this condition and remove this phase from `scala_library_for_plugin_bootstrapping`?

Reply:
Sounds like a great solution to me

Reply:
Fixed.